charts/openstack-cluster/templates/node-group/machine-deployment.yaml

```diff
@@ -47,7 +47,7 @@ spec:
         {{ $.Values.projectPrefix }}/node-group: {{ $nodeGroup.name }}
     spec:
       clusterName: {{ include "openstack-cluster.clusterName" $ }}
-      version: v{{ trimPrefix "v" $.Values.kubernetesVersion }}
+      version: v{{ $.Values.nodeGroupDefaults.kubernetesVersion | default $.Values.kubernetesVersion | trimPrefix "v" }}
```

Comment:
This should be `$nodeGroup.kubernetesVersion` instead of `$.Values.nodeGroupDefaults.kubernetesVersion`.

This will pick up the value from `nodeGroupDefaults` if not overridden in the node group, but also allows having a separate image per node group if desired.